backon/src/backoff/exponential.rs

```diff
@@ -128,6 +130,21 @@ impl ExponentialBuilder {
         self.max_times = None;
         self
     }
+
+    /// Set the total delay for the backoff.
+    ///
+    /// The backoff will stop yielding sleep durations once the cumulative sleep time
+    /// plus the next sleep duration would exceed `total_delay`.
+    pub const fn with_total_delay(mut self, total_delay: Duration) -> Self {
```

Comment:
total_delay is the new API, I believe we can just use `Option<Duration>` instead.